pkg/vault/contracts/Vault.sol

```diff
@@ -526,7 +526,7 @@ contract Vault is IVaultMain, VaultCommon, Proxy {
         //
         // Sets all fields in `poolData`. Side effects: updates `_poolTokenBalances`, and
         // `_aggregateFeeAmounts` in storage.
-        PoolData memory poolData = _loadPoolDataUpdatingBalancesAndYieldFees(params.pool, Rounding.ROUND_UP);
+        PoolData memory poolData = _loadPoolDataUpdatingBalancesAndYieldFees(params.pool, Rounding.ROUND_DOWN);
```

Comment:
There is actually a justification for this in the comment above.
Strictly speaking, to the best of my understanding rounding up is the correct thing to do here. The question is whether this actually makes any difference or not when we apply minimum amounts.